cmake/setup_cached_variables.cmake

```diff
@@ -370,13 +371,19 @@ UNSET(ENV{NVCCFLAGS})
 #                                                                      #
 ########################################################################
 
-
+include(CMakeDependentOption)
 OPTION(DEAL_II_WITH_64BIT_INDICES
   "If set to ON, then use 64-bit data types to represent global degree of freedom indices. The default is to OFF. You only want to set this to ON if you will solve problems with more than 2^31 (approximately 2 billion) unknowns. If set to ON, you also need to ensure that both Trilinos and/or PETSc support 64-bit indices."
   OFF
   )
 LIST(APPEND DEAL_II_FEATURES 64BIT_INDICES)
 
+CMAKE_DEPENDENT_OPTION(DEAL_II_USE_KOKKOS_BACKEND
+	"Experimental: If set to ON, Kokkos is used for all operations in LinearAlgebra::distributed::Vector. In particular, this allows accessing all Kokkos backends."
+  OFF "DEAL_II_WITH_KOKKOS" OFF
+  )
+LIST(APPEND DEAL_II_FEATURES KOKKOS_BACKEND)
```

Comment:
I know that this is not semantically elegant, but would you mind renaming the variable to `DEAL_II_WITH_KOKKOS_BACKEND`?

The reason being that this entire feature mechanism (i.e., when you add stuff to `DEAL_II_FEATURES` only works with the prefix `DEAL_II_WITH`). For example, a clean configure of your pull request currently shows:
```
#        DEAL_II_WITH_KOKKOS set up with external dependencies
#      ( DEAL_II_WITH_KOKKOS_BACKEND =  )
```